Source/SteveSloth/Private/MyPlayer.cpp

```diff
@@ -266,6 +310,37 @@ void AMyPlayer::MoveLeftRight(const FInputActionValue& Value)
 	// Add Animations here with changing of mesh direction
 }
 
+void AMyPlayer::GrapplingHook(const FInputActionValue& Value)
+{
+	if (bIsAimMode && bIsGrapplingHookUnlocked)
+	{
+		if (bDidGrapple == false)
+		{
+			UE_LOG(LogTemp, Warning, TEXT("Grappling Time!!"))
+		}
+
+		bDidGrapple = true;
+		// USE TIMER TO RESET GRAPPLING HOOK
+	}
+}
+
+void AMyPlayer::ClimbingClaw(const FInputActionValue& Value)
+{
+}
+
+void AMyPlayer::Shovel(const FInputActionValue& Value)
+{
+}
+
+void AMyPlayer::Propeller(const FInputActionValue& Value)
+{
+}
+
+void AMyPlayer::Magnetic(const FInputActionValue& Value)
+{
+}
+
+
```

Comment:
all these functions need to have functionality in them. As stated above these functions should be called from within a switch statement based on which upgrade is currently active.